src/hotspot/os/posix/signals_posix.hpp

```diff
@@ -25,45 +25,45 @@
 #ifndef OS_POSIX_SIGNALS_POSIX_HPP
 #define OS_POSIX_SIGNALS_POSIX_HPP
 
-#include "memory/allocation.hpp"
-#include "utilities/globalDefinitions.hpp"
+#include "memory/allStatic.hpp"
 
-#include <signal.h>
+// Forward declarations to be independent of the include structure.
 
-// Signal number used to suspend/resume a thread
-// do not use any signal number less than SIGSEGV, see 4355769
-static int SR_signum = SIGUSR2;
+typedef siginfo_t siginfo_t;
+typedef sigset_t sigset_t;
```

Comment:
I don't see why this is needed/wanted. We can include signal.h without a problem.

I'm not even sure what these typedefs means ??

Reply:
Coleen asked me to remove <signal.h> from the signals_posix.hpp, so those are forward declarations for the signal types we use.

I thought it was a reasonable request to minimize the number of headers. I saw some efforts in the past to cleanup header files, which is supposed to help with build times, so every little bit helps.

Reply:
Hi Gerard,

the proper way to do this would be to remove the header and forward declare the structures like this:

```
struct sigset_t;
struct siginfo_t;
struct ucontext_t;
```
That means we can use pointers to those things without including their definition.

But in this case I believe you could just remove signal.h without forward declaring anything. System headers are usually included via globalDefinitions.hpp, which we do include here. See globalDefinitions_gcc.hpp in this case.

Cheers, Thomas

Reply:
But its actually a matter of taste; leaving the forward declarations explicitly in like described above would be okay too.